historian-server/src/main/java/com/hurence/webapiservice/http/api/ingestion/ImportRequestParser.java

```diff
@@ -78,21 +94,59 @@ public ResponseAndErrorHolder checkAndBuildValidHistorianImportRequest(JsonArray
                 newTimeserie.put(POINTS_REQUEST_FIELD, newPoints);
                 responseAndErrorHolder.correctPoints.add(newTimeserie);
             }
+            int currentNumberOfFailedPoints;
+            JsonArray groupByArray = new JsonArray();
+            if (multiMap != null) {
+                List<String> groupByList = multiMap.getAll(GROUP_BY).stream().map(s -> {
+                    if (s.startsWith(TAGS+".")) {
+                        return s.substring(5);
+                    }else if (s.equals(NAME))
+                        return s;
+                    else return null ;
+                }).collect(Collectors.toList());
+                groupByList.remove(null);
+                groupByList.forEach(i -> groupByArray.add(timeserie.getValue(i)));
+            } else {
+                groupByArray.add(NAME);
+            }
+            if (responseAndErrorHolder.numberOfFailedPointsPerMetric.containsKey(groupByArray.toString())) {
+                currentNumberOfFailedPoints = responseAndErrorHolder.numberOfFailedPointsPerMetric.getInteger(groupByArray.toString());
+                responseAndErrorHolder.numberOfFailedPointsPerMetric.put(groupByArray.toString(), currentNumberOfFailedPoints+numberOfFailedPointsForThisName);
+            }else {
+                responseAndErrorHolder.numberOfFailedPointsPerMetric.put(groupByArray.toString(), numberOfFailedPointsForThisName);
+            }
+
         }
         if (responseAndErrorHolder.correctPoints.isEmpty())
             throw new IllegalArgumentException("There is no valid points");
         return responseAndErrorHolder;
-
     }
 
     public static class ResponseAndErrorHolder {
 
         public JsonArray correctPoints;
         public ArrayList<String> errorMessages;
+        public JsonObject numberOfFailedPointsPerMetric;
 
         ResponseAndErrorHolder () {
             this.correctPoints = new JsonArray();
             this.errorMessages = new ArrayList<>();
+            this.numberOfFailedPointsPerMetric = new JsonObject();
+        }
+    }
+    public static class ResponseAndErrorHolderAllFiles {
```

Comment:
why declaring this as a nested static class of ImportRequestParser ? This is not even used in ImprotRequestParser class.